docs/analyzer-rules/AZFW0002.md

```diff
@@ -0,0 +1,51 @@
+# AZFW0002: Avoid async void
+
+| | Value |
+|-|-|
+| **Rule ID** |AZFW0002|
+| **Category** |[Usage]|
+| **Severity** |Error|
+
+
+## Cause
+
+This rule is triggered when an async method uses `void` as return type.
+
+## Rule description
+
+When an exception is thrown out of an `async Task` or `async Task<T>` method, that exception is captured and placed on the `Task` object. With `async void` methods, there is no `Task` object, so any exceptions thrown out of an async void method will be posted to thread pool. An unhandled exception on the thread pool will cause the whole process to terminate.
```

Comment:
This verbiage seems to position exceptions as the main issue associated with a failure to return a `Task`, but the absence of a `Task` impacts more than the ability to handle and deal with exceptions. Without a `Task`, the infrastructure has nothing that provides a representation of the ongoing function execution and completion status, so a number of features may be impacted, causing reliability issues even for functions executing without exceptions.

Reply:
How about something like:

> An `async void` method does not provide a `Task` to represent the operation in progress, preventing the Azure Functions Worker and Host from tracking function completion and errors, leading to potential reliability.

Thoughts?

Reply:
Yea, This language is much cleaner. Switched to that. Thank you!